.github/workflows/automerge.yml

```diff
@@ -5,11 +5,35 @@
 # This reduces the friction associated with updating with our workflows.
 
 on: [ pull_request ]
+name: Automerge
 
 jobs:
-  automerge:
+  automerge-check:
     if: github.event.pull_request.user.login == 'web3-bot'
     runs-on: ubuntu-latest
+    outputs:
+      status: ${{ steps.should-automerge.outputs.status }}
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Check if we should automerge
+      id: should-automerge
+      run: |
+        for commit in $(git rev-list --first-parent origin/${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.sha }}); do
+          committer=$(git show --format=$'%ce' -s $commit)
+          echo "Committer: $committer"
+          if [[ "$committer" != "web3-bot@users.noreply.github.com" ]]; then
```

Comment:
I _think_ emails can be spoofed in commits. I'm curious do we need to worry about it?

Reply:
FYI this is just copied automatically from https://github.com/protocol/.github. It's not a security measure, but rather just a sanity check to catch some human mistakes. @marten-seemann can perhaps expand on this, but you should raise any issues on that repo :)

Reply:
Gotcha. I am curious is there a reason for combining PRs into one?

Reply:
to get CI to pass, that's all merged now so this should be just the new bits

Reply:
@masih See https://github.com/protocol/.github/pull/85#discussion_r625235315.